packages/common/core-interfaces/src/jsonString.ts

```diff
@@ -0,0 +1,85 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports -- incorrect rule: misunderstands `declare`d types.
+import type { BrandedType } from "./brandedType.js";
+import type { JsonDeserialized } from "./jsonDeserialized.js";
+import type { JsonSerializable } from "./jsonSerializable.js";
+
+/**
+ * Brand for JSON that has been stringified.
+ *
+ * Usage: Intersect with another type to apply branding.
+ *
+ * @sealed
+ */
+declare class JsonStringBrand<T> extends BrandedType<JsonString<unknown>> {
+	public toString(): string;
+	protected readonly EncodedValue: T;
+	private constructor();
+}
+
+/**
+ * Branded `string` for JSON that has been stringified.
+ *
+ * @remarks
+ *
+ * Use {@link JsonStringify} to encode JSON producing values of this type and
+ * {@link JsonParse} to decode them.
+ *
+ * For custom encoding/decoding:
+ *
+ * - cast to with `as unknown as JsonString<T>` when value of type `T` has been stringified.
+ *
+ * - use a form of {@link JsonDeserialized} for safety when parsing.
+ *
+ * @sealed
+ * @internal
+ */
+export type JsonString<T> = string & JsonStringBrand<T>;
+
+/**
+ * Options for {@link JsonStringify}.
+ *
+ * @internal
+ */
+export interface JsonStringifyOptions {
```

Comment:
Looks like this is duplicated from JsonSerializableOptions?

To avoid duplicating documentation, maybe defined this using `Pick<JsonSerializableOptions, "IgnoreInaccessibleMembers">`, then apply my suggested docs changes to the origional?